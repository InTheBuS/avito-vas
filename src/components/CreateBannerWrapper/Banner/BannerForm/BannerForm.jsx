import React from "react"
import {InputItem} from "./BannerInputItem/BannerInputItem";
import className from "./BannerForm.module.css"

export const BannerForm = ({fields = [], changeHandler}) => {

    const inputs = fields.length > 0
        ? fields.map((field, index) => {
        return (
            <InputItem key={index} field={field} changeHandler={changeHandler} index={index}/>
        )})
        :  <div>Что-то пошло не так</div>

    return (
        <div className={className.formWrapper}>
            {inputs}
        </div>
    )
}