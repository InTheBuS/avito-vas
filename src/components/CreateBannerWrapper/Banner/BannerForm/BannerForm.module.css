.formWrapper {
    display: flex;
    align-items: center;
    flex-wrap: wrap;
    width: 100%;
}
.formWrapper input {
    border: none;
    border-bottom: 1px solid black;
    outline: none;
}
.formWrapper input:hover {
    opacity: .6;
}

@media (min-width: 1400px) {
    .formWrapper {
        max-width: 900px;
        justify-content: center;
        font-size: 18px;
        gap: 40px 15px;
    }
    .formWrapper input {
        font-size: 16px;
    }
}
@media (max-width: 1399px) {
    .formWrapper {
        max-width: 900px;
        justify-content: center;
        font-size: 16px;
        margin: 25px 0;

    }
    .formWrapper input {
        font-size: 14px;
    }
}