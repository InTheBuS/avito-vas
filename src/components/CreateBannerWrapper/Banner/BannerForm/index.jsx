import React from "react"
import className from "./index.module.css"
import FieldGenerator from "../../../FieldGenerator";

export const BannerForm = ({fields = [], changeHandler}) => {

    const inputs = fields.length > 0
        ? fields.map((field, index) => {
            return FieldGenerator(field, index, changeHandler)
        })
        : null

    return (
        <div className={className.formWrapper}>
            {inputs}
        </div>
    )
}