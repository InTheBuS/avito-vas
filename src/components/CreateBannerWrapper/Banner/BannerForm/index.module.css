.formWrapper {
    width: 100%;
}
.formWrapper input {
    border: none;
    border-bottom: 1px solid black;
    outline: none;
}
.formWrapper input:hover {
    opacity: .6;
}

@media (min-width: 1400px) {
    .formWrapper {
        max-width: 900px;
        justify-content: center;
        font-size: 18px;
        display: flex;
        margin-top: 200px;
        flex-direction: column;
        align-self: flex-start;
    }
    .formWrapper input {
        font-size: 16px;
        width: 385px;
    }
}
@media (min-width: 900px) and (max-width: 1399px) {
    .formWrapper {
        max-width: 600px;
        justify-content: center;
        font-size: 18px;
        display: flex;
        margin-top: 200px;
        flex-direction: column;
        align-self: flex-start;
    }
    .formWrapper input {
        font-size: 16px;
        width: 200px;
    }
}
@media (max-width: 899px) {
    .formWrapper {
        max-width: 500px;
        justify-content: center;
        font-size: 14px;
        margin: 25px 0;

    }
    .formWrapper input {
        font-size: 14px;
        margin-right: 5px;
    }
}


