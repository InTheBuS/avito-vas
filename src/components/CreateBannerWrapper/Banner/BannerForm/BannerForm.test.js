import React from "react"
import renderer from "react-test-renderer"

import {BannerForm} from "./BannerForm";

describe("Testing BannerForm component", () => {

    let testArrayWithOneElem
    let testArrayWithMultipleElements

    beforeEach(() => {
        testArrayWithOneElem = [{name: "body", value: "This is the body text", type: "text", label: "test label", placeholder: "test placeholder"},]

        testArrayWithMultipleElements = [
            {name: "body", value: "This is the body text", type: "text", label: "test label", placeholder: "test placeholder"},
            {name: "color", value: "#646464", type: "color", label: "test label", placeholder: "test placeholder"},
            {name: "image", value: "", type: "text", label: "test label", placeholder: "test placeholder"},
        ]
    })

    it("rendering when array of inputs empty", () => {
        const tree = renderer.create(<BannerForm fields={testArrayWithOneElem}/>).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it("rendering when array has one object", () => {
        const tree = renderer.create(<BannerForm fields={testArrayWithMultipleElements}/>).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it("rendering when array has multiple objects", () => {
        const tree = renderer.create(<BannerForm />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})