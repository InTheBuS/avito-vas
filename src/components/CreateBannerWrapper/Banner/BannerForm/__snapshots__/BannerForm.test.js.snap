// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing BannerForm component rendering when array has multiple objects 1`] = `
<div
  className="formWrapper"
>
  <div>
    Что-то пошло не так
  </div>
</div>
`;

exports[`Testing BannerForm component rendering when array has one object 1`] = `
<div
  className="formWrapper"
>
  <label
    className="name"
  >
     
    test label:
    <input
      className="input"
      name="body"
      onChange={[Function]}
      placeholder="test placeholder"
      type="text"
      value="This is the body text"
    />
  </label>
  <label
    className="name"
  >
     
    test label:
    <input
      className="input"
      name="color"
      onChange={[Function]}
      placeholder="test placeholder"
      type="color"
      value="#646464"
    />
  </label>
  <label
    className="name"
  >
     
    test label:
    <input
      className="input"
      name="image"
      onChange={[Function]}
      placeholder="test placeholder"
      type="text"
      value=""
    />
  </label>
</div>
`;

exports[`Testing BannerForm component rendering when array of inputs empty 1`] = `
<div
  className="formWrapper"
>
  <label
    className="name"
  >
     
    test label:
    <input
      className="input"
      name="body"
      onChange={[Function]}
      placeholder="test placeholder"
      type="text"
      value="This is the body text"
    />
  </label>
</div>
`;
